### ISSUE TITLE
Missing Semicolon passes linting

### ISSUE THREAD

**Issue by jbcpollak** [eslint/typescript-eslint-parser#417]
I'm not sure if this is a problem with eslint, the parser, or the eslint-plugin-typescript project. Sometimes we find basic internal eslint rules don't fire correctly. In this case, the `semi` rule wasn't triggered correctly.

**What version of TypeScript are you using?**

typescript@2.6.1 

**What version of `typescript-eslint-parser` are you using?**

typescript-eslint-parser@9.0.0 

Other related dependency versions:

    "eslint": "^4.11.0",
    "eslint-config-6river": "^1.0.3",
    "eslint-plugin-mocha": "^4.11.0",
    "eslint-plugin-typescript": "^0.8.0",

**What code were you trying to parse?**

In a .ts file:

```
const j = Math.floor(Math.random() * idx)
```

In a .js file within the same project I correctly see the following:

` 194:40  error  Missing semicolon  semi`

**What did you expect to happen?**

I was expecting eslint to fail with an error that a semicolon was missing

**What happened?**

eslint passed correctly. Here is my config:

```
❯ ./node_modules/.bin/eslint --print-config .eslintrc.json
{
  "globals": {
    "__dirname": false,
    "__filename": false,
    "arguments": false,
    "Buffer": false,
    "clearImmediate": false,
    "clearInterval": false,
    "clearTimeout": false,
    "console": false,
    "exports": true,
    "GLOBAL": false,
    "global": false,
    "Intl": false,
    "module": false,
    "process": false,
    "require": false,
    "root": false,
    "setImmediate": false,
    "setInterval": false,
    "setTimeout": false,
    "Array": false,
    "ArrayBuffer": false,
    "Boolean": false,
    "constructor": false,
    "DataView": false,
    "Date": false,
    "decodeURI": false,
    "decodeURIComponent": false,
    "encodeURI": false,
    "encodeURIComponent": false,
    "Error": false,
    "escape": false,
    "eval": false,
    "EvalError": false,
    "Float32Array": false,
    "Float64Array": false,
    "Function": false,
    "hasOwnProperty": false,
    "Infinity": false,
    "Int16Array": false,
    "Int32Array": false,
    "Int8Array": false,
    "isFinite": false,
    "isNaN": false,
    "isPrototypeOf": false,
    "JSON": false,
    "Map": false,
    "Math": false,
    "NaN": false,
    "Number": false,
    "Object": false,
    "parseFloat": false,
    "parseInt": false,
    "Promise": false,
    "propertyIsEnumerable": false,
    "Proxy": false,
    "RangeError": false,
    "ReferenceError": false,
    "Reflect": false,
    "RegExp": false,
    "Set": false,
    "String": false,
    "Symbol": false,
    "SyntaxError": false,
    "System": false,
    "toLocaleString": false,
    "toString": false,
    "TypeError": false,
    "Uint16Array": false,
    "Uint32Array": false,
    "Uint8Array": false,
    "Uint8ClampedArray": false,
    "undefined": false,
    "unescape": false,
    "URIError": false,
    "valueOf": false,
    "WeakMap": false,
    "WeakSet": false,
    "after": false,
    "afterEach": false,
    "before": false,
    "beforeEach": false,
    "context": false,
    "describe": false,
    "it": false,
    "mocha": false,
    "run": false,
    "setup": false,
    "specify": false,
    "suite": false,
    "suiteSetup": false,
    "suiteTeardown": false,
    "teardown": false,
    "test": false,
    "xcontext": false,
    "xdescribe": false,
    "xit": false,
    "xspecify": false
  },
  "env": {
    "node": true,
    "es6": true,
    "mocha": true
  },
  "rules": {
    "no-cond-assign": 0,
    "no-irregular-whitespace": 2,
    "no-unexpected-multiline": 2,
    "valid-jsdoc": [
      2,
      {
        "requireParamDescription": false,
        "requireReturnDescription": false,
        "requireReturn": false,
        "prefer": {
          "returns": "return"
        }
      }
    ],
    "curly": [
      2,
      "multi-line"
    ],
    "guard-for-in": 2,
    "no-caller": 2,
    "no-extend-native": 2,
    "no-extra-bind": 2,
    "no-invalid-this": 2,
    "no-multi-spaces": 2,
    "no-multi-str": 2,
    "no-new-wrappers": 2,
    "no-throw-literal": 2,
    "no-with": 2,
    "no-unused-vars": [
      "off",
      {
        "args": "none"
      }
    ],
    "array-bracket-newline": 0,
    "array-bracket-spacing": [
      2,
      "never"
    ],
    "array-element-newline": 0,
    "block-spacing": [
      2,
      "never"
    ],
    "brace-style": 2,
    "camelcase": [
      2,
      {
        "properties": "never"
      }
    ],
    "comma-dangle": [
      2,
      "always-multiline"
    ],
    "comma-spacing": 2,
    "comma-style": 2,
    "computed-property-spacing": 2,
    "eol-last": 2,
    "func-call-spacing": 2,
    "key-spacing": 2,
    "keyword-spacing": 2,
    "linebreak-style": 2,
    "max-len": [
      "error",
      120,
      2
    ],
    "new-cap": 2,
    "no-array-constructor": 2,
    "no-mixed-spaces-and-tabs": [
      "error",
      "smart-tabs"
    ],
    "no-multiple-empty-lines": [
      2,
      {
        "max": 2
      }
    ],
    "no-new-object": 2,
    "no-tabs": "off",
    "no-trailing-spaces": 2,
    "object-curly-spacing": 2,
    "one-var": [
      2,
      {
        "var": "never",
        "let": "never",
        "const": "never"
      }
    ],
    "padded-blocks": [
      2,
      "never"
    ],
    "quote-props": [
      2,
      "consistent"
    ],
    "quotes": [
      2,
      "single",
      {
        "allowTemplateLiterals": true
      }
    ],
    "require-jsdoc": [
      "off",
      {
        "require": {
          "FunctionDeclaration": true,
          "MethodDefinition": true,
          "ClassDeclaration": true
        }
      }
    ],
    "semi": 2,
    "semi-spacing": 2,
    "space-before-blocks": 2,
    "space-before-function-paren": [
      2,
      {
        "asyncArrow": "always",
        "anonymous": "never",
        "named": "never"
      }
    ],
    "spaced-comment": [
      2,
      "always"
    ],
    "switch-colon-spacing": 2,
    "arrow-parens": [
      2,
      "always"
    ],
    "constructor-super": 2,
    "generator-star-spacing": [
      2,
      "after"
    ],
    "no-new-symbol": 2,
    "no-this-before-super": 2,
    "no-var": [
      "error"
    ],
    "prefer-rest-params": 2,
    "prefer-spread": 2,
    "rest-spread-spacing": 2,
    "yield-star-spacing": [
      2,
      "after"
    ],
    "prefer-const": [
      "error"
    ],
    "indent": [
      "error",
      "tab",
      {
        "MemberExpression": 0
      }
    ],
    "dot-location": [
      "error",
      "property"
    ],
    "no-console": [
      "error"
    ],
    "no-undef": [
      "off"
    ],
    "strict": [
      "error",
      "global"
    ],
    "object-shorthand": "error",
    "no-await-in-loop": "error",
    "eqeqeq": "error",
    "mocha/no-exclusive-tests": "error",
    "typescript/no-unused-vars": "error",
    "typescript/no-triple-slash-reference": "error"
  },
  "parserOptions": {
    "ecmaFeatures": {
      "globalReturn": true
    },
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "extends": "6river/typescript",
  "plugins": [
    "mocha",
    "typescript"
  ],
  "parser": "/Users/jpollak/src/task_coordinator/node_modules/typescript-eslint-parser/parser.js"
}
```

**Comment by JamesHenry** [comment]
Thanks for reporting, @jbcpollak! I'm afraid there is a bit too much at play here for this to be set up as an integration test within the repo itself, please could you provide a minimal reproduction of the issue?

**Comment by JamesHenry** [comment]
I'm going to close this due to lack of activity.

We now have a brand new way of running integration tests for this project, which allows for complex setups to be reproduced in isolation: https://github.com/eslint/typescript-eslint-parser#integration-tests

If this is still an issue for you using the latest version, please create a failing integration test for it and open a new issue to discuss.

Thanks!
